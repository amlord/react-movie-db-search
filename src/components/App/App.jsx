import React from 'react';
import styled from 'styled-components';
import theMovieDb from 'api/theMovieDb';

const App = () => {
  // Create a Title component that'll render an <h1> tag with some styles
  const Title = styled.h1`
    font-size: 1.5em;
    text-align: center;
    color: palevioletred;
  `;

  // Create a Wrapper component that'll render a <section> tag with some styles
  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `;

  theMovieDb.getMovies('fight c');

  return (
    <Wrapper>
      <Title>Base React App!</Title>
    </Wrapper>
  );
};

export default App;
