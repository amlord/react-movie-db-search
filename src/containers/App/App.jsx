import React, { Component } from 'react';
import Header from 'components/Header';
import SearchResults from 'containers/SearchResults';
import Footer from 'components/Footer';
import TheMovieDb from 'api/TheMovieDb';

class App extends Component {
  constructor(props) {
    super(props);

    this.TheMovieDb = new TheMovieDb();

    this.state = {
      results: [],
    };
  }

  /**
   * Search the movies api
   *
   * @param {String} searchText - text to search for
   * @returns {undefined}
   */
  searchMovies = (searchText) => {
    this.TheMovieDb.getMovies(searchText).then((response) => {
      this.setState({ ...response });
    });
  }

  /**
   * Handle search text change (prevent unnecessary searches)
   *
   * @param {String} searchText - text to search for
   * @returns {undefined}
   */
  handleSearchUpdate = (searchText) => {
    // clear previous timeout
    clearTimeout(this.timeout);

    // prevent search if no search text
    if (searchText.length === 0) {
      this.setState({ results: [], total_results: 0 });
    } else {
      // prevent firing off too many fetches in short succession
      this.timeout = setTimeout(() => {
        this.searchMovies(searchText);
      }, 250);
    }
  }

  render() {
    const { results, total_results: totalResults } = this.state;

    return (
      <>
        <Header />
        <SearchResults
          results={results}
          totalResults={totalResults}
          onSearchUpdate={this.handleSearchUpdate}
        />
        <Footer />
      </>
    );
  }
}

export default App;
