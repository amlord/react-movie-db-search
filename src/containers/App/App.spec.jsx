import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

// mock API object responses
jest.mock('api/TheMovieDb', () => (() => ({
  getMovies: () => Promise.resolve({
    results: [],
  }),
})));

describe('App', () => {
  it('displays correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
