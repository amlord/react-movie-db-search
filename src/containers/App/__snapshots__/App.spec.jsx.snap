// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchResults
          onSearchUpdate={[Function]}
          results={Array []}
          totalResults={0}
        />,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onSearchUpdate": [Function],
          "results": Array [],
          "totalResults": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchResults
            onSearchUpdate={[Function]}
            results={Array []}
            totalResults={0}
          />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSearchUpdate": [Function],
            "results": Array [],
            "totalResults": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
